ui/app/models/sync/destinations/vercel-project.js

```diff
@@ -36,12 +47,27 @@ export default class SyncDestinationsVercelProjectModel extends SyncDestinationM
   })
   teamId;
 
-  // TODO can also be a string, return and test how this works with live API
-  @attr('array', {
+  // comma separated string, updated as array using deploymentEnvironmentsArray
+  @attr({
     subText: 'Deployment environments where the environment variables are available.',
     editType: 'checkboxList',
-    possibleValues: ['deployment', 'preview', 'production'],
-    defaultValue: () => [],
+    possibleValues: ['development', 'preview', 'production'],
+    fieldValue: 'deploymentEnvironmentsArray', // getter/setter used to update value
   })
   deploymentEnvironments;
+
+  // Instead of using the 'array' attr transform, we keep deploymentEnvironments a string to leverage Ember's changedAttributes()
+  // which only tracks updates to string types. However, arrays are easier for managing multi-option selection so
+  // the fieldValue is used to get/set the deploymentEnvironments attribute to/from an array
+  get deploymentEnvironmentsArray() {
+    // if undefined or an empty string, return empty array
+    if (!this.deploymentEnvironments) return [];
+    return this.deploymentEnvironments.split(',');
+  }
+
+  set deploymentEnvironmentsArray(value) {
+    if (isArray(value)) {
+      this.deploymentEnvironments = value.join(',');
+    }
```

Comment:
I realize this is likely just a safe guard but maybe throwing an error might be better? If for some reason the value is a string for example then it won't change and remain set at the previous value which could be tricky to debug.

Reply:
Okay!